.github/workflows/ci.yml

```diff
@@ -187,1048 +128,35 @@ jobs:
           python3 -m venv .venv
           .venv/Scripts/activate.bat
           python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
+      - name: "Installing requirements"
+        run: choco install ccache --yes
       - name: "Configuring MSVC"
         uses: ilammy/msvc-dev-cmd@7315a94840631165970262a99c72cfb48a65d25d # v1.12.0
+      # Finally: build and run tests.
       - name: "Building IREE"
+        env:
+          IREE_READ_REMOTE_CCACHE: 0
+          IREE_WRITE_REMOTE_CCACHE: 0
+          IREE_READ_LOCAL_CCACHE: 1
+          IREE_WRITE_LOCAL_CCACHE: 1
+          # IREE_WRITE_LOCAL_CCACHE: ${{ needs.setup.outputs.write-caches }}
```

Comment:
Presumably you're going to flip this back? Reminder to do so

Reply:
Yeah. I'm troubleshooting the cache right now. A build earlier got a cache hit on the 2GB data but then had 100% cache misses... creating a 4GB cache entry. Assuming something like cache compression changes invalidated the cache (and not cosmic rays / other compiler settings), it might still make sense to just set the max cache size in ccache to 2-2.5GB.

Reply:
This is weirddd...

I changed a few comments and turned off `IREE_WRITE_LOCAL_CCACHE` and got
```
+ ccache --show-stats
Cacheable calls:   4007 / 4053 (98.87%)
  Hits:            1030 / 4007 (25.71%)
    Direct:        1025 / 1030 (99.51%)
    Preprocessed:     5 / 1030 ( 0.49%)
  Misses:          2977 / 4007 (74.29%)
Uncacheable calls:   46 / 4053 ( 1.13%)
Local storage:
  Cache size (GB): 2.42 / 3.00 (80.69%)
```

I'll try with `IREE_WRITE_LOCAL_CCACHE` on now... this is tricky to debug, but it will be even harder after it is merged.

Reply:
Maybe a significant number of cache hits are from the build compiling the same thing multiple times?

Reply:
Same behavior with writes enabled
```
+ ccache --show-stats
Cacheable calls:   3996 / 4053 (98.59%)
  Hits:            1030 / 3996 (25.78%)
    Direct:        1024 / 1030 (99.42%)
    Preprocessed:     6 / 1030 ( 0.58%)
  Misses:          2966 / 3996 (74.22%)
Uncacheable calls:   57 / 4053 ( 1.41%)
Local storage:
  Cache size (GB): 2.75 / 3.00 (91.75%)
  Cleanups:          41
```

I'm tempted to land this and watch the postsubmit stats. Presubmit will probably be too limited by cache misses given the size of the cache and low hit rates even with no changes...